### PR TITLE
NO-TICKET :: NCTL :: Enhanced compilation & logging.

### PR DESCRIPTION
**Pull request by ghost**
1. Added `NCTL_COMPILE_TARGET` flag to modify `nctl-compile` command.  Defaults to `release` but developer can set it to `debug` in order to compile `casper-node, casper-node-launcher & casper-client` in debug mode.  

2. Added `NCTL_NODE_LOG_FORMAT` flag.  Defaults to `json` but developer can set it to `text` in order to instruct `casper-node` to emit log messages in text format.  

3. Also prepared ground for adding delegators to accounts.toml at a later date.

### CODE REVIEW

File: utils/nctl/sh/assets/setup.sh [hunk 1]
```diff
@@ -133,6 +137,34 @@ validator_public_key = "${ACCOUNT_KEY_OF_VALIDATOR}"
 EOM
 }
 
+#######################################
+# Sets entry in chainspec's accounts.toml for a user account.
+# Arguments:
+#   Path to file containing a user's ed25519 public key in hex format.
+#   Path to file containing a validator's ed25519 public key in hex format.
```
Comment (Fraser999) [inline utils/nctl/sh/assets/setup.sh:144]: Is this correct?  Same question for the `Initial user staking weight.` 2 lines comment below.

**Review by Fraser999 — APPROVED** [review]
One non-blocking nitpick from me.  Thanks for the logging style support :)

**Review by goral09 — APPROVED** [review]
Looks good to me but It's really difficult to ~~eye-ball~~ review this. I trust that you have run it and verified it works.

File: utils/nctl/sh/assets/setup.sh (hunk 1 above)
Comment (momipsl) [reply utils/nctl/sh/assets/setup.sh:144]: Thx. I will review the function comments plus wider documentation.

**Comment by momipsl** [comment]
> Looks good to me but It's really difficult to ~eye-ball~ review this. I trust that you have run it and verified it works.

I have run it but I will run a deeper set of tests - i.e. joining + scenario tests, to smoke out any ci breaking issues.

**Pull request merged**